pennylane/tape/tape.py

```diff
@@ -349,6 +365,42 @@ def __exit__(self, exception_type, exception_value, traceback):
         finally:
             QuantumTape._lock.release()
 
+    @property
+    def circuit(self):
+        """Returns the quantum circuit recorded by the tape.
+
+        The circuit is created with the assumptions that:
+
+        * The ``operations`` attribute contains quantum operations and
```

Comment:
It is good to give this information. Do you think it would be desirable to defer all measurement to the end for the tape?

Reply:
>  Do you think it would be desirable to defer all measurement to the end for the tape?

That's what happens with the `defer_measurements` transform, right?

In terms of whether or not `tape.measurements` should contain both mid-circuit measurements and terminal measurements: that's a bit more complex (and likely out of scope for here). For now, it seems that this "hack" works well and we would not gain much from doing so in development.